### PR TITLE
Import sale order payments from magento

### PR DESCRIPTION
**Pull request by prakashpp**
Implements an intermediate model to map magento payment methods
to tryton payment gateways.
Import payments to tryton and complete them if they are already
completed in magento.

### CODE REVIEW

File: payment.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    _rec_name = 'title'
+
+    name = fields.Char("Name", required=True, select=True)
+    title = fields.Char('Title', required=True, select=True)
+    gateway = fields.Many2One(
+        'payment_gateway.gateway', 'Gateway', required=True,
+        ondelete='RESTRICT', select=True,
+    )
+    channel = fields.Many2One(
+        'sale.channel', 'Magento Channel', readonly=True
```
Comment (sharoonthomas) [inline payment.py]: select

File: payment.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    """
+    This model maps the available payment gateways from magento to tryton.
+    """
+    __name__ = 'magento.instance.payment_gateway'
+    _rec_name = 'title'
+
+    name = fields.Char("Name", required=True, select=True)
+    title = fields.Char('Title', required=True, select=True)
+    gateway = fields.Many2One(
+        'payment_gateway.gateway', 'Gateway', required=True,
```
Comment (sharoonthomas) [inline payment.py]: Add a domain to have manual only

File: payment.py (hunk 1 above)
Comment (prakashpp) [inline payment.py]: done

File: payment.py (hunk 2 above)
Comment (prakashpp) [inline payment.py]: I dont think we need it, next step we can introduce other gateways as well.

File: payment.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        ondelete='RESTRICT', select=True,
+    )
+    channel = fields.Many2One(
+        'sale.channel', 'Magento Channel', readonly=True, select=True,
```
Comment (tarunbhardwaj) [inline payment.py]: domain for magento channel only

File: payment.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    @classmethod
+    def __setup__(cls):
+        """
+        Setup the class before adding to pool
+        """
+        super(MagentoPaymentGateway, cls).__setup__()
+        cls._sql_constraints += [
+            (
+                'name_channel_unique', 'unique(name, channel)',
+                'Magento payment gateway must be unique in channel'
```
Comment (tarunbhardwaj) [inline payment.py]: better message for user.

`Payment gateway already exist for this channel.`

File: channel.py [hunk 5]
```diff
@@ -94,6 +94,9 @@ class Channel:
     product_listings = fields.One2Many(
         'product.product.channel_listing', 'channel', 'Product Listings',
     )
+    magento_payment_gateways = fields.One2Many(
```
Comment (tarunbhardwaj) [inline channel.py]: only visible for magento channel

File: view/sale_channel_form.xml [hunk 6]
```diff
@@ -41,6 +41,9 @@
                 <button string="Import Magento Carriers" name="import_magento_carriers"/>
             </group>
         </page>
+        <page string="Payment Gateways" id="payments">
```
Comment (tarunbhardwaj) [inline view/sale_channel_form.xml]: only visible for magento channels

File: payment.py (hunk 3 above)
Comment (prakashpp) [inline payment.py]: done

File: payment.py (hunk 4 above)
Comment (prakashpp) [inline payment.py]: done

File: view/sale_channel_form.xml (hunk 6 above)
Comment (prakashpp) [inline view/sale_channel_form.xml]: done

**Comment by tarunbhardwaj** [comment]
LGTM

**Pull request merged**